app/services/pipeline_report_service.rb

```diff
@@ -197,15 +248,27 @@ def compute_z_scores(taxa_counts, adjusted_total_reads)
       nr_z_score = compute_z_score(taxon_counts[:nr][:rpm], taxon_counts[:nr][:bg_mean], taxon_counts[:nr][:bg_stdev]) if taxon_counts[:nr].present?
       taxon_counts[:nt][:z_score] = nt_z_score if taxon_counts[:nt].present?
       taxon_counts[:nr][:z_score] = nr_z_score if taxon_counts[:nr].present?
+      taxon_counts[:nt][:z_score] = taxon_counts[:nt][:count] != 0 ? nt_z_score : Z_SCORE_WHEN_ABSENT_FROM_SAMPLE if taxon_counts[:nt].present?
+      taxon_counts[:nr][:z_score] = taxon_counts[:nr][:count] != 0 ? nr_z_score : Z_SCORE_WHEN_ABSENT_FROM_SAMPLE if taxon_counts[:nr].present?
       taxon_counts[:max_z_score] = nr_z_score.nil? || (nt_z_score && nt_z_score > nr_z_score) ? nt_z_score : nr_z_score
     end
   end
 
   def compute_aggregate_scores(species_counts, genus_counts)
     species_counts.each do |tax_id, species|
       genus = genus_counts[species[:genus_tax_id]]
-      species[:agg_score] = (species[:nt].present? ? genus[:nt][:z_score].abs * species[:nt][:z_score] * species[:nt][:rpm] : 0) \
-        + (species[:nr].present? ? genus[:nr][:z_score].abs * species[:nr][:z_score] * species[:nr][:rpm] : 0)
+      # Workaround placeholder for bad data (e.g. species counts present in TaxonSummary but genus counts aren't)
```

Comment:
Maybe add a TODO here to fix with a link to a JIRA ticket to fix the data issue?